packages/react-admin/src/mui/list/SingleFieldList.js

```diff
@@ -16,35 +21,106 @@ const styles = {
  *         <ChipField source="title" />
  *     </SingleFieldList>
  * </ReferenceManyField>
+ * 
+ * By default, it includes a link to the <Edit> page of the related record
+ * (`/books/:id` in the previous example).
+ *
+ * Set the linkType prop to "show" to link to the <Show> page instead.
+ *
+ * @example
+ * <ReferenceManyField reference="books" target="author_id" linkType="show">
+ *     <SingleFieldList>
+ *         <ChipField source="title" />
+ *     </SingleFieldList>
+ * </ReferenceManyField>
+ *
+ * You can also prevent `<SingleFieldList>` from adding link to children by setting
+ * `linkType` to false.
+ *
+ * @example
+ * <ReferenceManyField reference="books" target="author_id" linkType={false}>
+ *     <SingleFieldList>
+ *         <ChipField source="title" />
+ *     </SingleFieldList>
+ * </ReferenceManyField>
+
  */
-const SingleFieldList = ({
-    classes = {},
-    className,
-    currentSort,
-    ids,
-    isLoading,
-    data,
-    resource,
-    basePath,
-    children,
-    ...rest
-}) => (
-    <div className={classnames(classes.root, className)} {...rest}>
-        {ids.map(id =>
-            React.cloneElement(children, {
-                key: id,
-                record: data[id],
-                resource,
-                basePath,
-            })
-        )}
-    </div>
-);
+export class SingleFieldList extends Component {
+    // Our handleClick does nothing as we wrap the children inside a Link but it is
+    // required fo ChipField which uses a Chip from material-ui.
+    // The material-ui Chip requires an onClick handler to behave like a clickable element
+    handleClick = () => {};
+    render() {
+        const {
+            classes = {},
+            className,
+            ids,
+            data,
+            resource,
+            basePath,
+            children,
+            linkType,
+            ...rest
+        } = this.props;
+
+        return (
+            <div
+                className={classnames(classes.root, className)}
+                {...sanitizeRestProps(rest)}
+            >
+                {ids.map(id => {
+                    const href = linkToRecord(basePath, id);
```

Comment:
As it's repeated twice, couldn't you put the edit/show logic inside this `linkToRecord` helper function?